lib/api/v2/search_as_json.rb

```diff
@@ -33,17 +33,6 @@ def as_json_results_to_hash
     @results.collect { |result| as_json_result_hash result }
```

Comment:
Is this `as_json_result_hash` calling the method in `Api::V2::NonCommercialSearch`?

Reply:
@krbhavith Yes, good question.  I've moved the long remaining `as_json_result_hash` method from `NonCommercialSearch` to this module to make that fact even more obvious.  `CommercialSearch` itself is essentially deprecated as well, but still pops up in places related to the deprecated `ApiDocs` search functionality.  This was already a huge PR so I didn't want to take it any further, but I'm going to confirm there are still investigation tickets around covering the future removal of all that code as well (at which point this should become even simpler).  Thanks.